addons/l10n_br_pix/models/res_partner_bank.py

```diff
@@ -0,0 +1,123 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import re
+
+from odoo import models, fields, api, _
+from odoo.addons.mail.tools.mail_validation import mail_validate
+from odoo.exceptions import ValidationError
+from odoo.tools import float_repr
+
+
+class ResPartnerBank(models.Model):
+    _inherit = "res.partner.bank"
+
+    proxy_type = fields.Selection(
+        selection_add=[
+            ("email", "Email Address"),
+            ("mobile", "Mobile Number"),
+            ("br_cpf_cnpj", "CPF/CNPJ (BR)"),
+            ("br_random", "Random Key (BR)"),
+        ],
+        ondelete={
+            "email": "set default",
+            "mobile": "set default",
+            "br_cpf_cnpj": "set default",
+            "br_random": "set default",
+        },
+    )
+
+    @api.constrains("proxy_type", "proxy_value", "partner_id")
+    def _check_br_proxy(self):
+        for bank in self.filtered(lambda bank: bank.country_code == "BR" and bank.proxy_type != "none"):
+            if bank.proxy_type not in ("email", "mobile", "br_cpf_cnpj", "br_random"):
+                raise ValidationError(
+                    _(
+                        "The proxy type must be Email Address, Mobile Number, CPF/CNPJ (BR) or Random Key (BR) for Pix code generation."
+                    )
+                )
+
+            value = bank.proxy_value
+            if bank.proxy_type == "email" and not mail_validate(value):
+                raise ValidationError(_("%s is not a valid email.", value))
+
+            if bank.proxy_type == "br_cpf_cnpj" and (
+                not self.partner_id.check_vat_br(value) or any(not char.isdigit() for char in value)
+            ):
+                raise ValidationError(_("%s is not a valid CPF or CNPJ.", value))
+
+            if bank.proxy_type == "mobile" and (not value or not value.startswith("+55") or not len(value) != 14):
+                raise ValidationError(
+                    _(
+                        "The mobile number %s is invalid. It must start with +55, contain a 2 digit territory or state code followed by a 9 digit number.",
+                        value,
+                    )
+                )
+
+            regex = r"%(char)s{8}-%(char)s{4}-%(char)s{4}-%(char)s{4}-%(char)s{12}" % {"char": "[a-fA-F0-9]"}
+            if bank.proxy_type == "br_random" and not re.fullmatch(regex, bank.proxy_value):
+                raise ValidationError(
+                    _(
+                        "The random key %s is invalid, the format looks like this: 71d6c6e1-64ea-4a11-9560-a10870c40ca2",
+                        value,
+                    )
+                )
+
+    @api.depends("country_code")
+    def _compute_display_qr_setting(self):
+        """Override."""
+        bank_br = self.filtered(lambda b: b.country_code == "BR")
+        bank_br.display_qr_setting = self.env.company.qr_code
```

Comment:
We actually made a mistake by doing this, no other QR hides it's settings based on that field.
We changed it on other EMV in master to allow the POS to use them without activating the feature in accounting: https://github.com/odoo/odoo/pull/148803/commits/4f5e5416ca3f58746dc50660063ffe8785c20d42

Reply:
Ok, I changed it to `True` instead.